src/cmd-copy-container

```diff
@@ -0,0 +1,132 @@
+                    copies[f'{args.src_repo}@{digest}'] = f'{args.dest_repo}:{final_tag}'
+
+        for pullspec, pushspec in copies.items():
+            skopeo_copy(pullspec, args.authfile, pushspec, args.dest_authfile,
+                        args.v2s2)
+
+
+def skopeo_inspect(fqin, authfile):
+    args = ['skopeo', 'inspect', '--raw']
+    if authfile:
+        args += ['--authfile', authfile]
+    return run_get_json(args + [f'docker://{fqin}'])
+
+
+def skopeo_copy(pullspec, src_authfile, pushspec, dest_authfile, v2s2):
+    args = ['skopeo', 'copy', '--all', '--quiet']
+    if src_authfile and dest_authfile:
+        args += ['--src-authfile', src_authfile,
+                 '--dest-authfile', dest_authfile]
+    # assume --authfile applies to both src and dest
+    elif src_authfile:
+        args += ['--authfile', src_authfile]
+    elif dest_authfile:
+        args += ['--dest-authfile', dest_authfile]
+    if v2s2:
+        args += ['--format=v2s2', '--remove-signatures']
+    runcmd(args + [f'docker://{pullspec}', f'docker://{pushspec}'])
+
+
+# XXX: dedupe with oscontainer-deprecated-legacy-format.py
+def run_get_json(args):
+    return json.loads(runcmd(args, capture_output=True).stdout)
+
+
+def registry_supports_manifest_lists(repo):
+    # XXX: Ideally here, we'd figure out a way to query the registry to know if
+    # manifest lists are supported. For now, just hardcode known cases.
+    if repo.startswith("quay.io/"):
+        return True
+    if repo.startswith("registry.ci.openshift.org/"):
+        return False
+    # assume manifest lists are supported
+    return True
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        prog="cosa copy-container",
+        usage="%(prog)s [OPTION...] --tag=TAG ... SRC_REPO DEST_REPO",
+        description="Copy a container from one location to another.",
+        epilog=EXAMPLE_USAGE,
+        formatter_class=argparse.RawDescriptionHelpFormatter)
+
+    parser.add_argument("--authfile", help="A file to use for registry auth")
+    parser.add_argument("--dest-authfile",
+                        help="A file to use for dest registry auth")
+    # could support a `--tag SRC_TAG:DEST_TAG` syntax in the future if needed
+    parser.add_argument("--tag", required=True, dest='tags', action='append',
+                        help="The tag of the manifest to use")
+    parser.add_argument('--v2s2', action='store_true',
+                        help='Use old image manifest version 2 schema 2 format')
+    parser.add_argument("--manifest-list-to-arch-tag",
+                        choices=["always", "never", "auto"], default="auto",
+                        help="""Whether source images using manifest lists are
+                        converted to use `-${arch}` tag suffixes in the
+                        destination repo. `auto` enables the feature only if the
+                        destination registry doesn't support manifest lists.""")
```

Comment:
I wonder if - for our limited use cases - we should just make this not optional.

After October FCOS won't need to copy the container to a second location and the only use of it for RHCOS requires this IIUC.

Reply:
Are you suggesting we always require callers to pass `--manifest-list-to-arch-tag` to choose, or that the default should be `always`?

Reply:
Just an idea, but I'm thinking that we just don't support an argument at all.

In my mind this goes away when 4.11 is EOL and I don't want to grow more consumers of this than our particular use case.

We could leave the code in the initial commit and drop it in a secondary one. That way if we ever wanted it back it's in git history.

Reply:
Hmm, not sure. That switch is pretty central to this command's existence. IOW, once 4.11 goes EOL we could even drop this command entirely and just do vanilla `skopeo copy`s.

Reply:
> Hmm, not sure. That switch is pretty central to this command's existence. IOW, once 4.11 goes EOL we could even drop this command entirely and just do vanilla `skopeo copy`s.

This is exactly my point. Why make the differentiating feature optional? Unknowing future users might just come in and specify `--manifest-list-to-arch-tag=never` not knowing that they could just delete this code at that point and use `skopeo copy`.

Reply:
It feels odd to make it required if what the pipeline will want is `auto` anyway. Unless you're also considering surfacing it to `config.yaml`. I'm not too concerned about other users TBH. We could make this an "unlisted command" instead and type out `/usr/lib/coreos-assembler/...` in the pipeline if you prefer.

Reply:
Maybe there's a fundamental misunderstanding that I have. 

> It feels odd to make it required if what the pipeline will want is `auto` anyway.

After October (when we drop the FCOS 2nd container location) will the pipeline need to support both?

Reply:
It will not. But I'd prefer to keep the pipeline code generic. It's not exactly an RHCOS thing but rather a registry thing. I'd prefer the `registry.ci` detection hack to live in cosa.

Reply:
OK. Maybe after Oct we can re-visit.